src/main/java/org/opentripplanner/middleware/controllers/api/UserController.java

```diff
@@ -49,4 +81,18 @@ boolean preDeleteHook(OtpUser user, Request req) {
         }
         return true;
     }
+
+    /**
+     * HTTP endpoint to get the {@link OtpUser} entity, if it exists, from an {@link Auth0UserProfile} attribute
+     * available from a {@link Request} (this is the case for '/api/secure/' endpoints).
+     */
+    private OtpUser getUserFromRequest(Request req, Response res) {
+        Auth0UserProfile profile = Auth0Connection.getUserFromRequest(req);
+        OtpUser user = profile.otpUser;
+
+        if (user == null) {
+            logMessageAndHalt(req, HttpStatus.NOT_FOUND_404,String.format(NO_USER_WITH_AUTH0_ID_MESSAGE, profile.user_id),null);
```

Comment:
Please have better formatting here with spaces after commas. Also, the error string doesn't seem right because there will be an Auth0 user, just not an otpUser.

Reply:
@evansiroky, A valid Auth0 user id does not guarantee existence of a corresponding `OtpUser` entity with the proper `auth0UserId` attribute, and the error case covers that. Example: First-time users who just signed up on Auth0 and have not entered their preferences.